src/emulator/net/include/net/state.h

```diff
@@ -0,0 +1,44 @@
+// Vita3K emulator project
+// Copyright (C) 2018 Vita3K team
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+#pragma once
+
+#include <map>
+#include <memory>
+#include <vector>
+
+#ifdef WIN32
+# include <winsock2.h>
+# include <Ws2tcpip.h>
+typedef SOCKET abs_socket;
+typedef int socklen_t;
+#else
+# include <unistd.h>
+# include <sys/socket.h>
+# include <netinet/in.h>
+# include <netdb.h>
+# include <arpa/inet.h>
+typedef uint32_t abs_socket;
```

Comment:
Hardly any of these includes need to be in this public header. int is a built-in type and uint32_t comes from `<cstdint>`. I'm a bit wary of accidentally including Windows headers into the host state header, because that will cause the whole emulator to take a lot longer to build.

Maybe we can just use an int instead of abs_socket?

Reply:
Windows uses SOCKET that's a struct as result for socket(). Don't get your point here, sorry.

Reply:
The point was just to remove as many includes as possible. Looks like we don't need any here on macOS, but we need to include something on Windows.

Sorry for being annoying about this.

Reply:
Without those includes, we don't have required definitions on unix for sockets prototypes.